website/docs/r/private_key.html.md

```diff
@@ -3,14 +3,13 @@ layout: "tls"
 page_title: "TLS: tls_private_key"
 sidebar_current: "docs-tls-resource-private-key"
 description: |-
-  Creates a PEM-encoded private key.
+  Creates a PEM (and OpenSSH) formatted private key.
 ---
 
 # tls\_private\_key
 
-Generates a secure private key and encodes it as PEM. This resource is
-primarily intended for easily bootstrapping throwaway development
-environments.
+Generates a secure private key and encodes it as [PEM (RFC 1421)](https://datatracker.ietf.org/doc/html/rfc1421).
+This resource is primarily intended for easily bootstrapping throwaway development environments.
 
 ~> **Important Security Notice** The private key generated by this resource will
```

Comment:
Another note I'd probably add is that you shouldn't reuse the same key (output of this resource) between OpenSSH and TLS -- generate unique keys for each. Off the top of my head I'm not sure if its possible, but regardless, you don't want to open up the possibility of a cross-protocol attack due to key reuse (where e.g., you sign a TLS handshake hash or X.509 leaf cert with your OpenSSH private key that can then be massaged by an attacker into the correct form for impersonation in TLS/X.509 validation).

Reply:
While this is very sound advice, doesn't it cross the line of "we give you the resource, then it's up to you what you do with it"?

I wouldn't want to turn the documentation of those resources into a "how to do cryptography with this stuff, correctly": it wouldn't fit and, additionally, it needs a proper expert to write it - I'm not one :D